### ISSUE TITLE
Add experimental multi-tenancy record schema to protocol

### ISSUE THREAD

**Issue by korthout** [camunda/zeebe#13520]
To unblock the teams building multi-tenancy support to the other C8 components, we want to expand our protocol schema, which will be available for later usage, specifically by adding the affected tenant(s) to the record value interfaces.

## Details
The new additions to the schema must be marked as experimental.

As this is only to unblock other teams, the value will not yet be filled with actual tenant ids. Instead, they will contain an empty string.

## Schema changes
The record values involved with the following RPCs need a way to specify a tenant:

- DeployResource RPC
- CreateProcessInstance RPC
- CreateProcessInstanceWithResult RPC
- EvaluateDecision RPC
- BroadcastSignal RPC
- PublishMessage RPC

The record values involved with the following RPCs need a way to specify one or more tenants:

- ActivateJobs RPC
- StreamJobs RPC (we can consider this one out of scope for this issue)

## Out of scope
Note that these changes don't yet reflect the tenants a client may have access to. This will be covered in another issue.

## Additional context
- [Internal details](https://docs.google.com/document/d/1qWx82v7yqlJaL4NRXNca2Uxk4NZnmgh0sd5LQFeobHc/edit?pli=1#heading=h.6nar7bl09aov)

## Unblocks
- https://github.com/camunda/operate/issues/4778
